cdap-security/src/main/java/co/cask/cdap/security/authorization/DefaultAuthorizationEnforcer.java

```diff
@@ -43,12 +49,20 @@
 
   private final AuthorizerInstantiator authorizerInstantiator;
   private final boolean propagatePrivileges;
+  private final Principal masterUser;
 
   @Inject
   DefaultAuthorizationEnforcer(CConfiguration cConf, AuthorizerInstantiator authorizerInstantiator) {
     super(cConf);
     this.propagatePrivileges = cConf.getBoolean(Constants.Security.Authorization.PROPAGATE_PRIVILEGES);
     this.authorizerInstantiator = authorizerInstantiator;
+    String masterPrincipal = SecurityUtil.getMasterPrincipal(cConf);
+    try {
+      this.masterUser = masterPrincipal == null ? null : new Principal(new KerberosName(masterPrincipal).getShortName(),
```

Comment:
If masterPrincipal is null then the authorization bypass won't happen, and will lead to CDAP malfunctioning. Let's throw an exception with a good message here when masterPrincipal is null.